### PR TITLE
Use DSL specific types for contexts.

### PR DESCRIPTION
**Pull request by ckmcd**
Create per DSL Surge Context objects wrapping the core Context object.  Although we don't use the context yet, I anticipate that it will want a DSL specific interface.  Introducing this now will avoid breaking the API later.

### CODE REVIEW

File: modules/command-engine/core/src/main/scala/surge/core/Context.scala [hunk 1]
```diff
@@ -2,11 +2,13 @@
...
 package surge.core
 
+import akka.actor.ActorRef
+
 import scala.concurrent.ExecutionContext
-import scala.concurrent.ExecutionContext.global
 
-object Context {
-  val noop: Context = new Context()
+private[surge] object Context {
```
Comment (jeffboutotte) [inline modules/command-engine/core/src/main/scala/surge/core/Context.scala:9]: If these are both going to be private now (which makes sense to me for the core one) we can move them to the `surge.internal` package

File: modules/command-engine/javadsl/src/main/scala/surge/javadsl/common/Context.scala [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import surge.core.{ Context => Core }
+
+trait Context {}
+
+object Context {
+  def apply(core: Core): Context = new ContextImpl(core)
+}
+
+class ContextImpl(private val core: Core) extends Context
```
Comment (jeffboutotte) [inline modules/command-engine/javadsl/src/main/scala/surge/javadsl/common/Context.scala:13]: Minor but I think `ContextImpl` can be internal and we can keep the `Context` trait as the public/backwards compatible interface

File: modules/command-engine/scaladsl/src/main/scala/surge/scaladsl/common/Context.scala [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import surge.core.{ Context => Core }
+
+trait Context {}
+
+object Context {
+  def apply(core: Core): Context = new ContextImpl(core)
+}
+
+class ContextImpl(private val core: Core) extends Context
```
Comment (jeffboutotte) [inline modules/command-engine/scaladsl/src/main/scala/surge/scaladsl/common/Context.scala:13]: Same here as the javadsl - the impl can be internal

File: modules/command-engine/core/src/main/scala/surge/internal/persistence/PersistentActor.scala [hunk 4]
```diff
@@ -199,6 +199,8 @@ class PersistentActor[S, M, R, E](
     super.preStart()
   }
 
+  private def surgeContext() = Context(businessLogic.executionContext, self)
```
Comment (jeffboutotte) [inline modules/command-engine/core/src/main/scala/surge/internal/persistence/PersistentActor.scala:202]: How would you see a team interacting with the `ActorRef` here? There's a lot it's doing under the hood and it's very internal at the moment so my only concern would be that we could break consumers using the ActorRef relatively easily without knowing

Comment (ckmcd) [reply modules/command-engine/core/src/main/scala/surge/internal/persistence/PersistentActor.scala:202]: I anticipate that we will expose functions that interact with actor, but I do not intend to expose the actor ref directly to the app.

For instance, something like

```
def scheduleAlarm(time: Instance) = ScheduleAlarm(time) ? actorRef
```

**Review by jeffboutotte — APPROVED** [review]

**Pull request merged**